src/qt/clientmodel.cpp

```diff
@@ -46,6 +56,10 @@ ClientModel::ClientModel(interfaces::Node& node, OptionsModel *_optionsModel, QO
 ClientModel::~ClientModel()
 {
     unsubscribeFromCoreSignals();
+
+    m_thread->quit();
+    m_thread->wait();
+    delete pollTimer;
```

Comment:
As discussed on IRC: I don't think deleting the timer in the main thread, after having moved it to another thread is safe. Qt objects need to be deleted in the thread that owns them. I think that that thread doesn't exist anymore here doesn't change that. You're not even supposed to hold on to the pointer.

I think you can do this before moving it to the thread to make sure it's cleaned up:
```
connect(&thread, &QThread::finished, pollTimer, &QTimer::deleteLater);
```
(like in `RPCConsole::startExecutor()`)

Reply:
Done.